scripts/eurostat/health_determinants/common/euro_stat.py

```diff
@@ -0,0 +1,343 @@
+        final_df['info'] = final_df['SV'] + final_df['geo']
+        # Adding Measurement Method based on a condition, whereever d is found
+        # in an observation. The corresponding measurement method for all the
+        # years of that perticular SV/Country is made as Definition Differs.
+        # Eg: 2014
+        #   country/ITA
+        #   Percent_AlcoholConsumption_Daily_In_Count_Person
+        #   14.1 d,
+        # so measurement method for both 2014 and 2019 years shall be made
+        # Definition Differs.
+        final_df['Measurement_Method'] = np.where(
+            final_df['info'].isin(u_rows),
+            'EurostatRegionalStatistics_DefinitionDiffers',
+            final_df['Measurement_Method'])
+        final_df.drop(columns=['info'], inplace=True)
+        for drop_obs_char in [':', ' ', 'u', 'd', 'c']:
+            final_df['observation'] = final_df['observation'].astype(
+                str).str.replace(drop_obs_char, '')
+        final_df['observation'] = pd.to_numeric(final_df['observation'])
+        final_df = final_df.replace({'geo': COUNTRY_MAP})
+        final_df = final_df.sort_values(by=['geo', 'SV'])
+        final_df['observation'].replace('', np.nan, inplace=True)
+        final_df.dropna(subset=['observation'], inplace=True)
+        self._df = final_df
+        final_df.to_csv(
+            self._cleaned_csv_file_path,
+            columns=['time', 'geo', 'SV', 'observation', 'Measurement_Method'],
+            index=False)
+        return self._df
+
+    def generate_mcf(self, df: pd.DataFrame = None) -> None:
+        """
+        This method generates MCF file w.r.t
+        dataframe headers and defined MCF template
+
+        Args:
+            df (pd.DataFrame) :
+
+        Returns:
+            None
+        """
+        # pylint: disable=R0914
+        if df is not None:
+            self._df = df
+
+        final_mcf_template = ""
+        sv_list = list(set(self._df["SV"].to_list()))
+        sv_list.sort()
+
+        for sv in sv_list:
+            self._sv_properties = self._sv_properties_template
+            self._sv_properties = dict.fromkeys(self._sv_properties, "")
+            sv_name = ""
+
+            sv_temp = sv.split("_In_")
+            denominator = "\nmeasurementDenominator: dcs:" + sv_temp[1]
+            sv_prop = sv_temp[0].split("_")
+            sv_prop.append("Among")
+            sv_prop.extend(sv_temp[1].split("_"))
+
+            for prop in sv_prop:
+                if prop in ["Count", "Person"]:
+                    continue
+                if prop in ["Percent"]:
+                    sv_name = sv_name + "Percentage "
+                else:
+                    sv_name = sv_name + prop + ", "
+
+                for p in self._sv_value_to_property_mapping.keys():
+                    if p in prop:
+                        self._sv_properties[self._sv_value_to_property_mapping[
+                            p]] = self._sv_properties_template[
```

Comment:
can use a local variable for `self._sv_value_to_property_mapping[p]` as it is used multiple times and will make the statement more readable.